R/install_sys_deps.R

```diff
@@ -0,0 +1,186 @@
+addToProfile = function(line, type = 'OS') {
+  paths = if (type == 'OS') c('.zshrc', '.bashrc', '.profile') else '.Rprofile'
+  paths = paste0('~/', paths)
+
+  for (path in paths) {
+    lines = if (file.exists(path)) readLines(path) else character()
+    if (!(line %in% lines)) write(line, path, append = TRUE)}
+
+  invisible()}
+
+
+installSraToolkit = function(installDir, addToPath = TRUE) {
+  sraPath = path.expand(file.path(installDir, 'sratoolkit', 'bin'))
+
+  # Check if sratoolkit already exists at location
+  if (dir.exists(sraPath)) {
+    cat('SRA Toolkit already installed, skipping...\n')
+
+  } else {
+    cat('Installing SRA Toolkit...\n')
+    # Change working directory
+    withr::local_dir(installDir)
+
+    # Determine latest available SRA Toolkit version
+    url = 'https://ftp-trace.ncbi.nlm.nih.gov/sra/sdk/'
+    raw = RCurl::getURL(url)
+    x = strsplit(raw, '\\n')[[1L]]
+    m = regexpr('[0-9]\\.[0-9]\\.[0-9]', x)
+    sraVersion = max(regmatches(x, m))
+
+    # Based on operating system, download correct version to correct path
+    sraOsTar = if (Sys.info()[['sysname']] == 'Darwin') 'mac64' else 'ubuntu64'
+    sraTar = glue('sratoolkit.{sraVersion}-{sraOsTar}.tar.gz')
+
+    # Download based on version and OS, unzip, then rename directory
+    withr::local_file(sraTar)
+    download.file(glue('{url}{sraVersion}/{sraTar}'), sraTar, quiet = TRUE)
+    utils::untar(sraTar)
+    . = file.rename(sub('\\.tar\\.gz', '', sraTar), 'sratoolkit')}
+
+  configPath = '~/.ncbi/user-settings.mkfg'
+  if (!file.exists(configPath)) {
+    if (!dir.exists(dirname(configPath))) dir.create(dirname(configPath))
+    uuidPath = withr::local_tempfile(fileext = '.txt')
+    download.file(
+      'https://www.uuidgenerator.net/api/version4', uuidPath, quiet = TRUE)
```

Comment:
Had no clue this was a thing and it's useful! However, if you wanted we could just check for `uuidgen` locally since I never had a problem with it already being installed (and it would be one less external dependency) but I also think it's fine as is!